scripts/abins/abinsdata.py

```diff
@@ -1,79 +1,74 @@
 
-        # This should live closer to the Loaders but for now it is the only place the dict is used.
-        ab_initio_loaders = {"CASTEP": abins.input.CASTEPLoader, "CRYSTAL": abins.input.CRYSTALLoader,
-                             "DMOL3": abins.input.DMOL3Loader, "GAUSSIAN": abins.input.GAUSSIANLoader,
-                             "VASP": abins.input.VASPLoader}
-
-        if ab_initio_program.upper() not in ab_initio_loaders:
+        if ab_initio_program.upper() not in all_loaders:
             raise ValueError("No loader available for {}: unknown program. "
                              "supported loaders: {}".format(ab_initio_program.upper(),
-                                                            ' '.join(ab_initio_loaders.keys())))
-        loader = ab_initio_loaders[ab_initio_program.upper()](input_ab_initio_filename=filename)
-        return loader.get_formatted_data()
-
-    def set(self, k_points_data=None, atoms_data=None):
-        """
+                                                            ' '.join(all_loaders.keys())))
+        loader = all_loaders[ab_initio_program.upper()](input_ab_initio_filename=filename)
+        data = loader.get_formatted_data()
+        return data
 
-        :param k_points_data: object of type KpointsData
-        :param atoms_data: object of type AtomsData
-        """
-
-        if isinstance(k_points_data, abins.KpointsData):
-            self._kpoints_data = k_points_data
-        else:
-            raise ValueError("Invalid type of k-points data.")
-
-        if isinstance(atoms_data, abins.AtomsData):
-            self._atoms_data = atoms_data
-        else:
-            raise ValueError("Invalid type of atoms data.")
-
-        self._data = {"k_points_data": k_points_data.extract(), "atoms_data": atoms_data.extract()}
+    def get_kpoints_data(self) -> KpointsData:
+        """Get vibration data mapped over k-points"""
+        return self._k_points_data
 
-    def get_kpoints_data(self):
-        return self._kpoints_data
-
-    def get_atoms_data(self):
+    def get_atoms_data(self) -> AtomsData:
+        """Get atomic structure data"""
         return self._atoms_data
 
-    def extract(self):
-        for k in self._data["k_points_data"]["atomic_displacements"]:
-            if self._data["k_points_data"]["atomic_displacements"][k].shape[0] != len(self._data["atoms_data"]):
-                raise ValueError("Abins data is inconsistent.")
-
-        return self._data
-
-    def __str__(self):
-        return "DFT data"
+    def _check_consistent_dimensions(self) -> None:
+        """Raise an error if atoms_data and k_points_data have different numbers of atoms"""
+        data = self.extract()
+        for k in data["k_points_data"]["atomic_displacements"]:
+            if data["k_points_data"]["atomic_displacements"][k].shape[0] != len(data["atoms_data"]):
+                raise ValueError("Abins data is inconsistent: number of atoms in structure does not match "
```

Comment:
Haha, I suppose it should. In any case I think it would be more informative to have something like "KpointsData and AtomsData are not consistent: number of atoms..."

Reply:
Done